testing/kokoro/system_tests.sh

```diff
@@ -44,25 +44,27 @@ export GOLANG_SAMPLES_E2E_TEST=""
 # allow files to be owned by a different user than our current uid.
 # Kokoro runs a double-nested container, and UIDs may not match.
 git config --global --add safe.directory $(pwd)
-GIT_CHANGES=$(git --no-pager diff --name-only main..HEAD)
+# Allow $GIT_CHANGES to be set in the env, enabling local testing of the change detection below.
+GIT_CHANGES=${GIT_CHANGES:-$(git --no-pager diff --name-only main..HEAD)}
```

Comment:
`main` here is using the current status of the `main` branch in your checkout - which is likely missing some of the newest changes to `origin/main`. you can make those changes disappear by doing a `git pull` on your `main` branch, then switching back to the PR branch.

This is not a problem in CI, where we're using a fresh clone of the repo.

We may be able to address this by using `origin/main` instead of `main`, but that's probably unnecessary for our usage, and i'd like to minimize the behavior change in this PR. If you think this is an important change, you're welcome to file an issue for it.